lib/beaker-pe/install/pe_utils.rb

```diff
@@ -376,33 +408,41 @@ def deploy_frictionless_to_master(host)
-            # to avoid errors creating the same node group when the beaker hosts file contains
-            # multiple hosts with the same platform
-            node_group = _console_dispatcher.get_node_group_by_name('Beaker Frictionless Agent')
-            if node_group.nil? || node_group.empty?
-              node_group = {}
-              node_group['name'] = "Beaker Frictionless Agent"
-              # Pin the master to the node
-              node_group['rule'] = [ "and",  [ '=', 'name', master.to_s ]]
-              node_group['classes'] ||= {}
-            end
+            # Add pe_repo packages to 'PE Master' group
+            # This change will alow us to test the standard workflow where we add pe_repo packages
+            # to PE Master group itself. This will also make it easy to download the packages to
+            # compilemasters(PE-22051)
+            # Previosuly, we were creating a new node group 'Beaker Frictionless Agent' and adding
+            # pe_repo packages to that group
+            #
+            node_group = _console_dispatcher.get_node_group_by_name('PE Master')
 
             # add the pe_repo platform class if it's not already present
-            if ! node_group['classes'].include?(klass)
-              node_group['classes'][klass] = {}
-
-              _console_dispatcher.create_new_node_group_model(node_group)
-              # The puppet agent run that will download the agent tarballs to the master can sometimes fail with
-              # curl errors if there is a network hiccup. Use beakers `retry_on` method to retry up to
-              # three times to avoid failing the entire test pipeline due to a network blip
-              retry_opts = {
-                :desired_exit_codes => [0,2],
-                :max_retries => 3,
-                # Beakers retry_on method wants the verbose value to be a string, not a bool.
-                :verbose => 'true'
-              }
-              retry_on(master, puppet("agent -t"), retry_opts)
+            if (node_group)
+              if( ! node_group['classes'].include?(klass))
+                node_group['classes'][klass] = {}
+                _console_dispatcher.create_new_node_group_model(node_group)
+
+                # The puppet agent run that will download the agent tarballs to the master can sometimes fail with
+                # curl errors if there is a network hiccup. Use beakers `retry_on` method to retry up to
+                # three times to avoid failing the entire test pipeline due to a network blip
+                retry_opts = {
+                  :desired_exit_codes => [0,2],
+                  :max_retries => 3,
+                  # Beakers retry_on method wants the verbose value to be a string, not a bool.
+                  :verbose => 'true'
+                }
+                retry_on(master, puppet("agent -t"), retry_opts)
+
+                # Download the agent tarballs to compile masters
+                if ! compile_masters.empty?
```

Comment:
@shaigy according to the beaker test run, this isn't defined and is causing the tests to fail at the moment; is there something different about your local setup that is including this compile_masters variable?

Reply:
I ran install.rb from pe_acceptance_test repo as below:
shaigy@shaigy-C02LX3NAFH05:~/repo/pe_acceptance_tests$ bundle exec beaker -h ~/beaker/beaker-pe.conf --pre-suite setup/install.rb --helper lib/beaker_helper.rb

I am wondering whether there is a gem difference or something. I will need to take a closer look

Reply:
It could be that the way the tests are running haven't been updated to run how you are currently running it; I'll double check that as well.

Reply:
I was wondering the same. Thanks Tony for looking into it.

Reply:
Tony, I couldn't find any difference between the gems used in my local testing and Jenkins PR testing. 
Should I just try the following change?
if ! compile_masters.empty?
to 
if ( ! compile_masters.nil? &&  ! compile_masters.empty?)